six/two/two.cpp

```diff
@@ -370,50 +370,71 @@ PyObject *Two::_findSubclassOf(PyObject *base, PyObject *module)
 
     PyObject *dir = PyObject_Dir(module);
     if (dir == NULL) {
+        PyErr_Clear();
         setError("there was an error calling dir() on module object");
         return NULL;
     }
 
-    PyObject *klass = NULL;
+    PyObject *klass;
     for (int i = 0; i < PyList_GET_SIZE(dir); i++) {
-        // get symbol name
-        char *symbol_name;
+        // Reset `klass` at every iteration so its state is always clean when we
+        // continue the loop or return early. Reset at first iteration is useless
+        // but it keeps the code readable.
+        klass = NULL;
+        Py_XDECREF(klass);
```

Comment:
you need to DECREF before setting to NULL ;)

Reply:
You probably don't want to `Py_XDECREF(klass)` on an uninitialized pointer though - thinking about the first iteration.